flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/calcite/FlinkPlannerImpl.scala

```diff
@@ -179,15 +154,7 @@ class FlinkPlannerImpl(
         schemaPath: util.List[String],
         viewPath: util.List[String]): RelRoot = {
 
-      val parser: SqlParser = SqlParser.create(queryString, parserConfig)
-      var sqlNode: SqlNode = null
-      try {
-        sqlNode = parser.parseQuery
-      }
-      catch {
-        case e: CSqlParseException =>
-          throw new SqlParserException(s"SQL parse failed. ${e.getMessage}", e)
-      }
+      val sqlNode: SqlNode = parser.parse(queryString)
```

Comment:
Is it intended to re-use parser here? I noticed in `PlannerBase::parser` you pointed out that you didn't want to reuse parser since config might changed.

Reply:
Yes, because the whole `FlinkPlannerImpl` is used only once per planning session.